src/treetensornetworks/projttns/projttn_apply.jl

```diff
@@ -1,57 +1,114 @@
 struct ProjTTNApply{V} <: AbstractProjTTN{V}
   pos::Union{Vector{<:V},NamedEdge{V}}
-  psi0::TTN{V}
-  H::TTN{V}
+  init_state::AbstractTTN{V}
+  operator::AbstractTTN{V}
```

Comment:
We shouldn't put abstract types as fields, I would revert this and we can generalize if needed in another way.